To: vim_dev@googlegroups.com
Subject: Patch 7.3.318
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.318
Problem:    "C" on the last line deletes that line if it's blank.
Solution:   Only delete the last line for a delete operation. (James Vega)
Files:	    src/ops.c


*** ../vim-7.3.317/src/ops.c	2011-08-10 12:10:55.000000000 +0200
--- src/ops.c	2011-09-21 17:32:36.000000000 +0200
***************
*** 1922,1928 ****
  		    curwin->w_cursor.coladd = 0;
  	    }
  #endif
! 	    if (oap->inclusive && oap->end.lnum == curbuf->b_ml.ml_line_count
  		    && n > (int)STRLEN(ml_get(oap->end.lnum)))
  	    {
  		/* Special case: gH<Del> deletes the last line. */
--- 1922,1930 ----
  		    curwin->w_cursor.coladd = 0;
  	    }
  #endif
! 	    if (oap->op_type == OP_DELETE
! 		    && oap->inclusive
! 		    && oap->end.lnum == curbuf->b_ml.ml_line_count
  		    && n > (int)STRLEN(ml_get(oap->end.lnum)))
  	    {
  		/* Special case: gH<Del> deletes the last line. */
***************
*** 3331,3338 ****
  	if (regname == '=')
  	{
  	    /* For the = register we need to split the string at NL
! 	     * characters. */
! 	    /* Loop twice: count the number of lines and save them. */
  	    for (;;)
  	    {
  		y_size = 0;
--- 3333,3340 ----
  	if (regname == '=')
  	{
  	    /* For the = register we need to split the string at NL
! 	     * characters.
! 	     * Loop twice: count the number of lines and save them. */
  	    for (;;)
  	    {
  		y_size = 0;
***************
*** 3348,3354 ****
  			if (y_array != NULL)
  			    *ptr = NUL;
  			++ptr;
! 			/* A trailing '\n' makes the string linewise */
  			if (*ptr == NUL)
  			{
  			    y_type = MLINE;
--- 3350,3356 ----
  			if (y_array != NULL)
  			    *ptr = NUL;
  			++ptr;
! 			/* A trailing '\n' makes the register linewise. */
  			if (*ptr == NUL)
  			{
  			    y_type = MLINE;
*** ../vim-7.3.317/src/version.c	2011-09-21 17:15:34.000000000 +0200
--- src/version.c	2011-09-21 17:31:48.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     318,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
26. You check your mail. It says "no new messages." So you check it again.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
